core/iwasm/interpreter/wasm_loader.c

```diff
@@ -925,6 +925,14 @@ resolve_value_type(const uint8 **p_buf, const uint8 *buf_end,
                                    error_buf, error_buf_size))
             return false;
         *p_need_ref_type_map = true;
+#if WASM_ENABLE_STRINGREF != 0
+        /* covert (ref string) to stringref */
```

Comment:
> Does `(ref string)` equal to stringref? For other reference types, xxxref equals to `(ref null xxx)` but not `(ref xxx)`, e.g., structref is same as `(ref null struct)` but not `(ref struct)`, it is nullable. My understanding is that stringref should comply with the rule, could you check whether stringref is nullable?

This is not well defined, but binaryen‘s optimizer sometimes optimize (stringref) to (ref string), and usually assign a (stringref) to (ref string), we've tested with V8 and these are treated as valid code, so we add such workaround here, otherwise we can't run the optimized wasm module